src/Headers/Hydrodynamics.h

```diff
@@ -84,6 +86,9 @@ class Hydrodynamics
   void CheckBoundaryGhostParticle(const int, const int, const FLOAT, const DomainBox<ndim> &);
 
   void CreateBoundaryGhostParticle(const int, const int, const int, const FLOAT, const FLOAT);
+  Particle<ndim>& CreateNewParticle(const enum ptype, const enum parttype, const int, const int, const int,
+                                    const FLOAT, const FLOAT, const FLOAT, const FLOAT*, const FLOAT*,SimulationBase*);
+
```

Comment:
I don't think there is any other way of doing it that does not force **the caller** to flag for the tree rebuild. And that is much more dangerous than passing this pointer, because who writes the caller will surely forget to set the flag